src/fsharp/MethodOverrides.fs

```diff
@@ -420,7 +421,83 @@ module DispatchSlotChecking =
+                                     |> List.sortWith (fun (ty1, _) (ty2, _) ->
+                                         if TypeFeasiblySubsumesType 0 g amap reqdTyRange ty1 CanCoerce ty2 then 1
+                                         else -1
+                                     )
+                                
+                                 match sortedSlots with
+                                 | [] ->
+                                    yield RequiredSlot(reqdSlot, not reqdSlot.IsAbstract)
+                                 | (ty1, (RequiredSlot (_, isOptional1) as bestSlot)) :: sortedSlotsTail ->
+
+                                    // TODO: Add check for most specific implementation when an explicit implementation is given.
+                                    let mutable hasMostSpecificImplementation = true
+                                    let mutable isOptional1 = isOptional1
+                                    for (ty2, RequiredSlot (_, isOptional2)) in sortedSlotsTail do
+
+                                        let isNotPartOfHierarchy =
+                                            not (TypeFeasiblySubsumesType 0 g amap reqdTyRange ty1 CanCoerce ty2) && 
+                                            not (TypeFeasiblySubsumesType 0 g amap reqdTyRange ty2 CanCoerce ty1)
+
+                                        if isOptional1 && isOptional2 && hasMostSpecificImplementation && isNotPartOfHierarchy then
+                                            hasMostSpecificImplementation <- false
+
+                                        if isNotPartOfHierarchy then
+                                            isOptional1 <- if isOptional1 then true else isOptional2 
+
+                                    if hasMostSpecificImplementation then
+                                        yield bestSlot
+                                    else
+                                        // TODO: Move this to SR
+                                        errorR(Error((5000, sprintf "Interface member %A does not have a most specific implementation." (NicePrint.stringOfMethInfo amap reqdTyRange denv reqdSlot)), reqdTyRange))
+                                        // We have already error'ed, so make this optional to not show many more errors.
+                                        yield RequiredSlot(reqdSlot, true)
+
+               else
+                   
+                   // In the normal case, the requirements for a class are precisely all the abstract slots up the whole hierarchy.
+                   // So here we get and yield all of those.
+                   for minfo in reqdTy |> GetIntrinsicMethInfosOfType infoReader None AccessibleFromSomewhere AllowMultiIntfInstantiations.Yes IgnoreOverrides reqdTyRange do
+                      if minfo.IsDispatchSlot then
+                          yield RequiredSlot(minfo, (*isOptional=*) not minfo.IsAbstract) ]
```

Comment:
Is the `(*isOptional=*)` comment necessary here?

Reply:
Probably not. This code existed before; I just moved it around.